platform/core/src/utils/combineFrameInstance.ts

```diff
@@ -9,34 +11,68 @@
-    instance;
+  const {
+    PerFrameFunctionalGroupsSequence,
+    SharedFunctionalGroupsSequence,
+    NumberOfFrames,
+    SpacingBetweenSlices,
+  } = instance;
 
   if (PerFrameFunctionalGroupsSequence || NumberOfFrames > 1) {
     const frameNumber = Number.parseInt(frame || 1);
-    const shared = (
-      SharedFunctionalGroupsSequence ? Object.values(SharedFunctionalGroupsSequence[0]) : []
-    )
-      .filter(it => !!it)
-      .map(it => it[0])
-      .filter(it => it !== undefined && typeof it === 'object');
-    const perFrame = (
-      PerFrameFunctionalGroupsSequence
-        ? Object.values(PerFrameFunctionalGroupsSequence[frameNumber - 1])
-        : []
-    )
-      .filter(it => !!it)
-      .map(it => it[0])
-      .filter(it => it !== undefined && typeof it === 'object');
+    const shared = SharedFunctionalGroupsSequence
+      ? Object.values(SharedFunctionalGroupsSequence[0])
+          .filter(Boolean)
+          .map(it => it[0])
+          .filter(it => it !== undefined && typeof it === 'object')
+      : [];
+
+    const perFrame = PerFrameFunctionalGroupsSequence
+      ? Object.values(PerFrameFunctionalGroupsSequence[frameNumber - 1])
+          .filter(Boolean)
+          .map(it => it[0])
+          .filter(it => it !== undefined && typeof it === 'object')
+      : [];
 
     // this is to fix NM multiframe datasets with position and orientation
     // information inside DetectorInformationSequence
     if (!instance.ImageOrientationPatient && instance.DetectorInformationSequence) {
       instance.ImageOrientationPatient =
         instance.DetectorInformationSequence[0].ImageOrientationPatient;
     }
+
+    let ImagePositionPatientToUse = instance.ImagePositionPatient;
+
     if (!instance.ImagePositionPatient && instance.DetectorInformationSequence) {
-      instance.ImagePositionPatient = instance.DetectorInformationSequence[0].ImagePositionPatient;
+      const imagePositionPatient = instance.DetectorInformationSequence[0].ImagePositionPatient;
+
+      // Calculate the position for the current frame
+      if (instance.ImageOrientationPatient && SpacingBetweenSlices) {
+        const rowOrientation = vec3.fromValues(
+          instance.ImageOrientationPatient[0],
+          instance.ImageOrientationPatient[1],
+          instance.ImageOrientationPatient[2]
+        );
+
+        const colOrientation = vec3.fromValues(
+          instance.ImageOrientationPatient[3],
+          instance.ImageOrientationPatient[4],
+          instance.ImageOrientationPatient[5]
+        );
+
+        const normalVector = vec3.create();
+        vec3.cross(normalVector, rowOrientation, colOrientation);
+
+        const position = vec3.create();
+        vec3.scaleAndAdd(
+          position,
+          imagePositionPatient,
+          normalVector,
+          SpacingBetweenSlices * (frameNumber - 1)
+        );
+
+        ImagePositionPatientToUse = [position[0], position[1], position[2]];
```

Comment:
You can cast this from position rather than creating a new vector.  It isn't QUITE well defined, but typescript allows it
That is ImagePositionPatientToUse = position as Types.Point3
is a valid assignment.